src/vs/workbench/contrib/files/browser/explorerService.ts

```diff
@@ -366,6 +380,25 @@ export class ExplorerService implements IExplorerService {
 		}
 	}
 
+	// Reveal excludes
+	private matchesUpToRoot(item: ExplorerItem | undefined): boolean {
+		if (item === undefined) {
+			return false;
+		}
+		const root = item.root;
+		let currentItem = item;
+		while (currentItem !== root) {
+			if (this.revealExcludeMatcher.matches(currentItem.resource, name => !!(currentItem.parent && currentItem.parent.getChild(name)))) {
```

Comment:
Looks like yiou don't actually need the Promise here?

Reply:
You're right, I will make it boolean only